src/rviz_publisher.cpp

```diff
@@ -0,0 +1,118 @@
+ *     contributors may be used to endorse or promote products derived from
+ *     this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+// %Tag(FULLTEXT)%
+#include "ros/ros.h"
+#include "std_msgs/String.h"
+#include <visualization_msgs/Marker.h>
+#include "iarc7_msgs/OrientationThrottleStamped.h"
+
+/**
+ * This tutorial demonstrates simple receipt of messages over the ROS system.
+ */
+// %Tag(CALLBACK)%
+class SubscribeAndPublish
+{
+public:
+  SubscribeAndPublish()
+  {
+    //Topic you want to publish
+    ros::Publisher marker_pub = n.advertise<visualization_msgs::Marker>("visualization_marker", 1);
+
+    //Topic you want to subscribe
+    ros::Subscriber sub = n.subscribe("uav_direction_command", 1000, &SubscribeAndPublish::sendToRviz, this);
+  }
+
+
+  void sendToRviz(const iarc7_msgs::OrientationThrottleStamped::ConstPtr& msg)
+  {
+    ROS_INFO("Pitch is %lf",msg->data.pitch);
+    visualization_msgs::Marker marker;
+    uint32_t shape = visualization_msgs::Marker::CUBE;
+    marker.header.frame_id = "/my_frame";
+    marker.header.stamp = ros::Time::now();
+    marker.ns = "basic_shapes";
+    marker.id = 0;
+    marker.type = shape;
+    marker.pose.position.x = 0;
+    marker.pose.position.y = 0;
+    marker.pose.position.z = 0;
+    double t0 = std::cos(msg->data.yaw * 0.5f);
+    double t1 = std::sin(msg->data.yaw * 0.5f);
+    double t2 = std::cos(msg->data.roll * 0.5f);
+    double t3 = std::sin(msg->data.roll * 0.5f);
+    double t4 = std::cos(msg->data.pitch * 0.5f);
+    double t5 = std::sin(msg->data.pitch * 0.5f);
+    marker.pose.orientation.w = t0 * t2 * t4 + t1 * t3 * t5;
+    marker.pose.orientation.x = t0 * t3 * t4 - t1 * t2 * t5;
+    marker.pose.orientation.y = t0 * t2 * t5 + t1 * t3 * t4;
+    marker.pose.orientation.z = t1 * t2 * t4 - t0 * t3 * t5;
+    marker.scale.x = 1.0;
+    marker.scale.y = 1.0;
+    marker.scale.z = 1.0;
+    marker.color.r = 0.0f;
+    marker.color.g = 1.0f;
+    marker.color.b = 0.0f;
+    marker.color.a = 1.0;
+    marker.lifetime = ros::Duration();
+
+    marker_pub.publish(marker);
+
+
+  }
+
+private:
+  ros::NodeHandle n; 
+  ros::Publisher marker_pub;
```

Comment:
Can we give these a name specific to the type of thing they are subscribing or publishing too? That will help when we start to add more publishers and subscribers.